server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -884,6 +889,18 @@ private void checkStatusOfCurrentlyExecutingBackups() {
         }
     }
 
+    @DB
+    public void cleanupBackups() {
+        LOG.debug("Disassociating backup offerings from expunged VMs that have no more backups");
+        List<VMInstanceVO> destroyedVmsWithOfferings = vmInstanceDao.listDestroyedVmsWithBackupOfferingAssigned();
+        for(VMInstanceVO instanceVO : destroyedVmsWithOfferings) {
+            List<Backup> backupsForVm = backupDao.listByVmId(instanceVO.getDataCenterId(), instanceVO.getId());
+            if (CollectionUtils.isEmpty(backupsForVm)) {
+                removeVMFromBackupOffering(instanceVO.getId(), true);
```

Comment:
@Pearl1594 should this be put in a try-catch; an exception will fail this and effectively the `poll`